src/main/scala/org/broadinstitute/dsde/workbench/sam/dataAccess/PostgresDistributedLockDAO.scala

```diff
@@ -49,17 +50,8 @@ class PostgresDistributedLockDAO[F[_]](
 
   private[dsde] def getLockStatus(lock: LockDetails): F[LockStatus] = {
     val res = for {
-      lockRecord <- OptionT(F.delay(retrieveLock(lock)))
-      expiresAt <- OptionT.fromOption[F](Option(lockRecord.expiresAt))
-      statusF = F.realTimeInstant.map[LockStatus] { currentTime =>
-        if (currentTime.isAfter(expiresAt)) {
-          deleteLock(lock)
-          Available
-        } else
-          Locked
-      }
-      status <- OptionT.liftF[F, LockStatus](statusF)
-    } yield status
+      _ <- OptionT(F.delay(retrieveLock(lock)))
+    } yield Locked
```

Comment:
the way I'm reading this, it will always yield Locked (unless something went wrong in the for comp) which doesn't seem right. Am I missing something here?

Reply:
I've modified the `retrieveLock` function to only return a `Some(Lock)` if one is available and unexpired. If it is expired, `retrieveLock` deletes it and returns `None`. If `retrieveLock` returns `None`, then the `OptionT` will not  reach the `yield` clause, as it short-circuits and `res` is `None`. The `fold` operation on res` then provides a default `Available`. If `res` is a `Some(Locked)`, then the `fold` operation will return the `Locked`.

This could be re-written as `OptionT(F.delay(retrieveLock(lock))).map(_ => Locked).fold[LockStatus](Available)(identity)`, or simplified even further to `F.delay(retrieveLock(lock).fold[LockStatus](Available)(_ => Locked))`.

In fact, why don't I just make it the simplified version and get rid of `OptionT`...